server/src/main/java/io/deephaven/server/console/completer/PythonAutoCompleteObserver.java

```diff
@@ -0,0 +1,222 @@
+                    String completionName = completion.getAttribute("name").getStringValue();
+                    int completionPrefix = completion.call("get_completion_prefix_length").getIntValue();
+                    final CompletionItem.Builder item = CompletionItem.newBuilder();
+                    final TextEdit.Builder textEdit = item.getTextEditBuilder();
+                    textEdit.setText(completionName);
+                    final DocumentRange.Builder range = textEdit.getRangeBuilder();
+                    int start = pos.getCharacter() - completionPrefix;
+                    item.setStart(start);
+                    item.setLabel(completionName);
+                    item.setLength(completionName.length());
+                    range.getStartBuilder().setLine(pos.getLine()).setCharacter(start);
+                    range.getEndBuilder().setLine(pos.getLine()).setCharacter(start + completionName.length());
+                    item.setInsertTextFormat(2);
+                    if (completionName.startsWith("__")) {
+                        completionResults__.add(item);
+                    } else if (completionName.startsWith("_")) {
+                        completionResults_.add(item);
+                    } else {
+                        completionResults.add(item);
+                    }
+                }
+                int sortPos = 0;
+                List<CompletionItem> finalItems = new ArrayList<>();
+                for (CompletionItem.Builder res : completionResults) {
+                    res.setSortText(ChunkerCompleter.sortable(sortPos++));
+                    finalItems.add(res.build());
+                }
+                for (CompletionItem.Builder res : completionResults_) {
+                    res.setSortText(ChunkerCompleter.sortable(sortPos++));
+                    finalItems.add(res.build());
+                }
+                for (CompletionItem.Builder res : completionResults__) {
+                    res.setSortText(ChunkerCompleter.sortable(sortPos++));
+                    finalItems.add(res.build());
+                }
+
+                final GetCompletionItemsResponse builtItems = GetCompletionItemsResponse.newBuilder()
+                        .setSuccess(true)
+                        .setRequestId(request.getRequestId())
+                        .addAllItems(finalItems)
+                        .build();
+
+                safelyExecuteLocked(responseObserver,
+                        () -> responseObserver.onNext(AutoCompleteResponse.newBuilder()
+                                .setCompletionItems(builtItems)
+                                .build()));
+                String totalNano = Long.toString(System.nanoTime() - startNano);
+                // lets track how long completions take, as it's known that some
+                // modules like numpy can cause slow completion, and we'll want to know what was causing them
+                log.info().append("Jedi completions took ")
```

Comment:
also, logging should be pulled out of here, since we probably a) don't want to count IO in the jedi time, and b) don't want to fail to log time if IO blew up

Reply:
cleaned up the logging and set it to TRACE.

Also, if I'm only logging > 100ms, I can probably make stronger guesses about string length and simplify the toString ugliness too...
Also, where should I move logging to make it not-count?
Also, who is counting / cares about how long this method takes?

I did the logging part after the work part completed, so if there is a better location / way to do this, you'll have to point me at it explicitly.